main/test/ca/uwaterloo/flix/library/Test.Exp.Struct.New.flix

```diff
@@ -0,0 +1,214 @@
+mod Mod1 {
+    pub struct S [r] {}
+    struct S2 [r] {}
+
+    mod Mod2 {
+        @test
+        def testOuterMod01(): Unit =
+            region rc {
+                new Mod1.S2 {} @ rc;
+                ()
+            }
+    }
+}
+
+mod Test.Exp.Struct.New {
+
+    struct Empty [r] { }
+
+    struct OneEmpty [r] {
+        a: Empty[r]
+    }
+
+    struct TwoEmpty [r] {
+        a: Empty[r],
+        b: Empty[r]
+    }
+
+    struct EmptyC [r] {
+        c: Empty[r]
+    }
+
+    struct EmptyB [r] {
+        b: EmptyC[r]
+    }
+
+    struct EmptyA [r] {
+        a: EmptyB[r]
+    }
+
+    struct A [r] {
+        a: B[r]
+    }
+
+    struct B [r] {
+        b: C[r]
+    }
+
+    struct C [r] {
+        c: D[r]
+    }
+
+    struct D [r] {
+        d: Int32
+    }
+
+    struct Name [r] {
+        name: String
+    }
+
+    struct FullName [r] {
+        fstName: String,
+        lstName: String
+    }
+
+    struct Person [r] {
+        fstName: String,
+        lstName: String,
+        age: Int32
+    }
+
+    struct Cowboy [r] {
+        fstName: String,
+        lstName: String,
+        age: Int32,
+        cowboy: Bool
+    }
+
+    struct TwoInts [r] {
+        x: Int32,
+        y: Int32
+    }
+
+    struct ThreeInts [r] {
+        x: Int32,
+        y: Int32,
+        z: Int32
+    }
+
+    struct Outer [r] {
+        a: TwoInts[r],
+        b: TwoInts[r]
+    }
+
+    struct TwoIntsHolder [r] {
+        a: TwoInts[r]
+    }
+
+    struct A3 [r] {
+        a: A2[r]
+    }
+
+    struct A2 [r] {
+        a: A1[r]
+    }
+
+    struct A1 [r] {
+        a: Empty[r]
+    }
+
```

Comment:
I think we need more tests that are highly recursive and polymorphic. It is always there the bugs are.